### DISCUSSION TITLE
Help with the Quality of image

### DISCUSSION THREAD

**Discussion by troelschristensen** [ForNeVeR/wpf-math#315] [Q&A] [answered]
Hi. 
I've been using your code on my site.
Right now I use the function latex2png to get a latex-image back in my other code. I'm coding in vb. Sorry. ;) It's working. But the quality is not the best. The scale defines how big the latex-image is, when it's beeing returned. 
Larger image (Scale) = better image quality.

But when I set a scale too low, the quality is almost unreadable. Hope you understand my issue and can help me get better quality.

Kind regards.

![image](https://user-images.githubusercontent.com/16334255/182677696-e3127c3c-64ea-41d8-8ac6-ce115bcda83a.png)
![image](https://user-images.githubusercontent.com/16334255/182677721-20348d47-0655-4a21-b2e1-59e8a3c11cf3.png)

```
Shared Function latex2png(Scale As Double, latex As String) As MemoryStream

        Dim parser = New TexFormulaParser()
        Dim formula = parser.Parse(latex)
        formula.GetRenderer(TexStyle.Text, Scale, "Calibri") 

        Dim pngBytes = formula.RenderToPng(Scale, 0.0, 0.0, "Calibri")
        Dim latexpng As New MemoryStream(pngBytes)

    Return latexpng
End Function
```

**Comment by ForNeVeR** [comment, marked as answer]
Try calling `renderer.RenderToBitmap(0, 0, DPI)`, where `DPI` is a number bigger than the standard 96 (say, 192).

When in doubt, check the code of `RenderToPng`; I believe it only uses public stuff and is very simplified.
